fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -547,17 +554,23 @@ private Query getFilterQuery(Set<ApiFilter> filters, int perPage) {
                 .map(document -> document.get(idKey))
                 .map(dimension::findDimensionRowByKeyValue)
                 .collect(Collectors.toCollection(TreeSet::new));
+
+        int documentCount;
+        lock.readLock().lock();
```

Comment:
Don't we need to get this lock much earlier in this method? We're getting `indexSearcher` up near the top of the method...